### ISSUE TITLE
Error autoForm insert requires doc argument when multiple schemas

### ISSUE THREAD

**Issue by gregory** [aldeed/meteor-autoform#1506]
TLDR: 
<img width="860" alt="screenshot 2016-08-17 00 33 14" src="https://cloud.githubusercontent.com/assets/167488/17728107/61da76a8-6412-11e6-966a-847e12ac38dd.png">

Here is what happens: 

I get an array of schemas attached to a collection, just [like in the doc](https://github.com/aldeed/meteor-collection2#attaching-multiple-schemas-to-the-same-collection). 
When trying to build a form here is the issue: 
`{{>quickForm id='myInsertForm' doc=doc schema=schema collection='myCollection' type='insert'}}`
If i try to submit, i get a `Error collection.simpleSchema() requires doc argument when there are multiple schemas` like shown above in the screenshot.

It [appear](https://github.com/aldeed/meteor-collection2-core/blob/master/lib/collection2.js#L118) that simpleSchema() does require at least a doc arguments. 

Here is the stacktrace: 

```
Error: collection.simpleSchema() requires doc argument when there are multiple schemas
    at obj.simpleSchema (http://localhost:3000/packages/aldeed_collection2-core.js?hash=78aa4a4dfcd4f54f1b080164dd6ca648f9f16f75:156:23)
    at Object.onSubmit (http://localhost:3000/packages/aldeed_autoform.js?hash=5dbf44ff89f182bd8c2512330e170ef4d5bf9582:3084:43)
    at Object.autoFormSubmitHandler (http://localhost:3000/packages/aldeed_autoform.js?hash=5dbf44ff89f182bd8c2512330e170ef4d5bf9582:7842:18)
    at Object.dict.(anonymous function) (http://localhost:3000/packages/kadira_debug.js?hash=b307db4b0f8441a4097f509b8f98f57bfa87c5c1:429:19)
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:3718:20
    at Function.Template._withTemplateInstanceFunc (http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:3687:12)
    at .<anonymous> (http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:3717:25)
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:2560:30
```

Another bug that is related to this issue is that the schema should be retrieved based on the doc & the collection, and as of now, this throws an exception:
`{{>quickForm id='myInsertForm' doc=doc  collection='myCollection' type='insert'}}`

<img width="1016" alt="screenshot 2016-08-17 00 46 36" src="https://cloud.githubusercontent.com/assets/167488/17728446/1e9a6202-6414-11e6-9643-2bb2bfbcd0b9.png">

Here is the full stacktrace: 

```
Error: Exception in template helper:
Error: collection.simpleSchema() requires doc argument when there are multiple schemas
    at obj.simpleSchema (http://localhost:3000/packages/aldeed_collection2-core.js?hash=78aa4a4dfcd4f54f1b080164dd6ca648f9f16f75:156:23)
    at setDefaults (http://localhost:3000/packages/aldeed_autoform.js?hash=5dbf44ff89f182bd8c2512330e170ef4d5bf9582:3015:29)
    at Object.AutoForm.parseData (http://localhost:3000/packages/aldeed_autoform.js?hash=5dbf44ff89f182bd8c2512330e170ef4d5bf9582:2771:10)
    at Object.quickFormContext (http://localhost:3000/packages/aldeed_autoform.js?hash=5dbf44ff89f182bd8c2512330e170ef4d5bf9582:6696:33)
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:2994:16
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:1653:16
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:3046:66
    at Function.Template._withTemplateInstanceFunc (http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:3687:12)
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:3045:27
    at Object.Spacebars.call (http://localhost:3000/packages/spacebars.js?hash=65db8b6a8e3fca189b416de702967b1cb83d57d5:172:18)
```

cc @aldeed

**Comment by jshimko** [comment]
Looks like a duplicate of https://github.com/aldeed/meteor-collection2-core/issues/6

**Comment by gregory** [comment]
@jshimko There is a PR opened that should fix it, hope it'll help :)
